packages/ember-data/tests/integration/adapter/json-api-adapter-test.js

```diff
@@ -875,3 +875,37 @@ test('update record - serialize hasMany', function() {
     });
   });
 });
+
+test('fetching a relationship via link works with data = null - #3451', function() {
+  expect(3);
+
+  ajaxResponse([{
+    data: {
+      type: 'post',
+      id: '1',
+      attributes: {
+        title: 'Ember.js rocks'
+      },
+      relationships: {
+        author: {
+          links: {
+            related: 'http://example.com/author/1'
```

Comment:
Can we change this URL to `http://example.com/post/1/author`? It really doesn't make sense to provide a URL to a specific author with an ID and then return `null`.

Sorry for the OCD.